spyder_kernels/console/kernel.py

```diff
@@ -540,67 +540,52 @@ def set_matplotlib_conf(self, conf):
-            )
-
+            rc.update({'figure.figsize': (conf[width_n], conf[height_n])})
         if fontsize_n in conf:
-            self._set_mpl_inline_rc_config('font.size', conf[fontsize_n])
-
+            rc.update({'font.size': conf[fontsize_n]})
         if bottom_n in conf:
-            self._set_mpl_inline_rc_config(
-                'figure.subplot.bottom',
-                conf[bottom_n]
-            )
+            rc.update({'figure.subplot.bottom': conf[bottom_n]})
+        if rc:
+            self._set_inline_config_option('rc', rc)
 
         if bbox_inches_n in conf:
-            self.set_mpl_inline_bbox_inches(conf[bbox_inches_n])
+            bbox_inches = 'tight' if conf[bbox_inches_n] else None
+            self._set_inline_config_option(
+                'print_figure_kwargs', {'bbox_inches': bbox_inches}
+            )
 
-
-    def set_mpl_inline_bbox_inches(self, bbox_inches):
-        """
-        Set inline print figure bbox inches.
-
-        The change is done by updating the 'print_figure_kwargs' config dict.
-        """
-        config = self.config
-        inline_config = (
-            config['InlineBackend'] if 'InlineBackend' in config else {})
-        print_figure_kwargs = (
-            inline_config['print_figure_kwargs']
-            if 'print_figure_kwargs' in inline_config else {})
-        bbox_inches_dict = {
-            'bbox_inches': 'tight' if bbox_inches else None}
-        print_figure_kwargs.update(bbox_inches_dict)
-
-        # This seems to be necessary for newer versions of Traitlets because
-        # print_figure_kwargs doesn't return a dict.
-        if isinstance(print_figure_kwargs, LazyConfigValue):
-            figure_kwargs_dict = print_figure_kwargs.to_dict().get('update')
-            if figure_kwargs_dict:
-                print_figure_kwargs = figure_kwargs_dict
-
-        self._set_config_option(
-            'InlineBackend.print_figure_kwargs', print_figure_kwargs)
+        # To update rcParams in inline mode we can either do so directly or
+        # re-assert inline mode. However, either will prevent restoring
+        # rcParams when toggling to interactive mode. The workaround is to
+        # first toggle to interactive mode, then back to inline. This updates
+        # the rcParams and restores rcParams when switching to interactive.
+        interactive_backend = self.get_mpl_interactive_backend()
+        current_backend = self.get_matplotlib_backend()
+        pylab_autoload_o = conf.get(pylab_autoload_n, False)
+        pylab_backend_o = conf.get(pylab_backend_n, current_backend)
+        backend_changed = current_backend != pylab_backend_o
+        if (
+            current_backend == 'inline' and rc
+            and not backend_changed
+            and interactive_backend not in ('inline', -1)
+        ):
+            self._set_mpl_backend(interactive_backend)
+        if (
+            (current_backend == 'inline' and rc)  # toggle back to inline
```

Comment:
```suggestion

        # Toggle back to inline
        if (
            (current_backend == 'inline' and rc)
```

Reply:
Toggling back to inline is only one of the reasons to run the code in this `if` block, not the only one. The other reasons are if autoloading pylab or the backend has changed.

Reply:
Nevertheless, this code block will certainly change given the comments above regarding the toggling "hack".